File: functions/src/models/MatchFilter.ts
/* eslint-disable */


import {BadModelFieldException} from "../exceptions/BadModelFieldException";

export class MatchFilter {

    private fromMembers!:number ;
    private toMembers!:number ;
    private enabled!:boolean ;
    private noTerrain!:boolean ;
    private yesTerrain!:boolean ;
    private maybeTerrain!:boolean;
    public static fromMembersConstant:number = 0;
    public static toMembersConstant:number = 15;


    constructor() {

    }

    public static makeInstance(snapshotNeedsParsing: any): MatchFilter{

        const filter:MatchFilter = new MatchFilter();

        let parsedSnapshot;
        try{
            parsedSnapshot = JSON.parse(snapshotNeedsParsing);
        }catch (Error){
            parsedSnapshot = snapshotNeedsParsing;
        }

        filter.setFromMembers(parsedSnapshot.fromMembers);
        filter.setEnabled(parsedSnapshot.enabled);
        filter.setToMembers(parsedSnapshot.toMembers);
        filter.setYesTerrain(parsedSnapshot.yesTerrain);
        filter.setMaybeTerrain(parsedSnapshot.maybeTerrain);
        filter.setNoTerrain(parsedSnapshot.noTerrain);

        filter.validateFromAndToMembers();

        return filter;
    }


    public getFromMembers(): number {
        return this.fromMembers;
    }

    public setFromMembers(fromMembers: number) {
        if (fromMembers === undefined || typeof fromMembers !== 'number' || isNaN(fromMembers)) {
            throw new BadModelFieldException("fromMembers must be a valid number");
        }


        if (fromMembers < MatchFilter.fromMembersConstant)
            throw new BadModelFieldException("fromMembers must be >= 0");


        this.fromMembers = fromMembers;
    }

    public getToMembers(): number {
        return this.toMembers;
    }

    public setToMembers(toMembers: number) {
        if (toMembers > MatchFilter.toMembersConstant)
            throw new BadModelFieldException("fromMembers must be <= " + MatchFilter.toMembersConstant);


        this.toMembers = toMembers;
    }

    public isEnabled(): boolean {
        return this.enabled;
    }

    public setEnabled(enabled: boolean) {
        if (enabled === undefined || typeof enabled !== 'boolean')
            throw new BadModelFieldException("enabled must be a valid boolean");


        this.enabled = enabled;
    }

    public isNoTerrain(): boolean {
        return this.noTerrain;
    }

    public setNoTerrain(noTerrain: boolean) {
        if (noTerrain === undefined || typeof noTerrain !== 'boolean')
            throw new BadModelFieldException("noTerrain must be a valid boolean");


        this.noTerrain = noTerrain;
    }

    public isYesTerrain(): boolean {
        return this.yesTerrain;
    }

    public setYesTerrain(yesTerrain: boolean) {
        if (yesTerrain === undefined || typeof yesTerrain !== 'boolean')
            throw new BadModelFieldException("yesTerrain must be a valid boolean");


        this.yesTerrain = yesTerrain;
    }

    public isMaybeTerrain(): boolean {
        return this.maybeTerrain;
    }

    public setMaybeTerrain(maybeTerrain: boolean) {
        if (maybeTerrain === undefined || typeof maybeTerrain !== 'boolean')
            throw new BadModelFieldException("maybeTerrain must be a valid boolean");


        this.maybeTerrain = maybeTerrain;
    }

    private validateFromAndToMembers() {

        if (this.fromMembers > this.toMembers)
            throw new BadModelFieldException("fromMembers is bigger than toMembers");

    }
}

File: app/src/main/java/com/baikas/sporthub6/models/MatchFilter.java
package com.baikas.sporthub6.models;

import java.io.Serializable;

public class MatchFilter implements Serializable {

    private int fromMembers;
    private int toMembers;
    private boolean noTerrain;
    private boolean yesTerrain;
    private boolean maybeTerrain;
    private boolean enabled;
    public static int fromMembersConstant = 1;
    public static int toMembersConstant = 15;

    public MatchFilter(int fromMembers, int toMembers, boolean enabled, boolean noTerrain, boolean yesTerrain, boolean maybeTerrain) {
        this.fromMembers = fromMembers;
        this.toMembers = toMembers;
        this.enabled = enabled;
        this.noTerrain = noTerrain;
        this.yesTerrain = yesTerrain;
        this.maybeTerrain = maybeTerrain;
    }

    public static MatchFilter resetFilterEnabled() {

        return new MatchFilter(fromMembersConstant,toMembersConstant,true,true,true,true);
    }

    public static MatchFilter resetFilterDisabled() {

        return new MatchFilter(fromMembersConstant,toMembersConstant,false,true,true,true);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        MatchFilter that = (MatchFilter) o;

        if (fromMembers != that.fromMembers) return false;
        if (toMembers != that.toMembers) return false;
        if (enabled != that.enabled) return false;
        if (noTerrain != that.noTerrain) return false;
        if (yesTerrain != that.yesTerrain) return false;
        return maybeTerrain == that.maybeTerrain;
    }

    @Override
    public int hashCode() {
        int result = fromMembers;
        result = 31 * result + toMembers;
        result = 31 * result + (noTerrain ? 1 : 0);
        result = 31 * result + (enabled ? 1 : 0);
        result = 31 * result + (yesTerrain ? 1 : 0);
        result = 31 * result + (maybeTerrain ? 1 : 0);
        return result;
    }

    public boolean isNoTerrain() {
        return noTerrain;
    }

    public void setNoTerrain(boolean noTerrain) {
        this.noTerrain = noTerrain;
    }

    public boolean isYesTerrain() {
        return yesTerrain;
    }

    public void setYesTerrain(boolean yesTerrain) {
        this.yesTerrain = yesTerrain;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public boolean isMaybeTerrain() {
        return maybeTerrain;
    }

    public void setMaybeTerrain(boolean maybeTerrain) {
        this.maybeTerrain = maybeTerrain;
    }

    public int getFromMembers() {
        return fromMembers;
    }

    public void setFromMembers(int fromMembers) {
        this.fromMembers = fromMembers;
    }

    public int getToMembers() {
        return toMembers;
    }

    public void setToMembers(int toMembers) {
        this.toMembers = toMembers;
    }
}
